packages/jest-circus/README.md

```diff
@@ -28,6 +28,8 @@ class MyCustomEnvironment extends NodeEnvironment {
 
 Mutating event or state data is currently unsupported and may cause unexpected behavior or break in a future release without warning. New events, event data, and/or state data will not be considered a breaking change and may be added in any minor release.
 
+Note, that `jest-circus` test runner would pause until a promise returned from `handleTestEvent` gets fulfilled. **However, there are a few events that do not conform to this rule, namely**: `start_describe_definition`, `finish_describe_definition`, `add_hook`, `add_test` or `error` (for the up-to-date list you can look at [SyncEvent type in the types definitions](https://github.com/facebook/jest/tree/master/packages/jest-types/src/Circus.ts)). That is caused by backward compatibility reasons and `process.on('unhandledRejection', callback)` signature, but that usually should not be a problem for most of the use cases.
```

Comment:
could we verify this in code?

e.g. if the event is any of those, throw an error if the handler returns a promise? I guess that would force people to not use `async` (or write some convoluted herler functions), which would suck, so probably not...

Reply:
> could we verify this in code?

I can try writing another test for that, though need to think how - looks challenging.

> or write some convoluted helper functions

Exactly, this is why I would rather refrain from throwing an error.

Reply:
yeah, let's drop it. I don't think we wanna disallow `async function`s here

Reply:
Great, thanks! See my recent comment about a warning assertion in the e2e test.